src/nvim/vvars.lua

```diff
@@ -63,6 +69,21 @@ M.vars = {
     ]=],
   },
   completed_item = {
+    class = [[
+--- @field word? string the text that will be inserted, mandatory
+--- @field abbr? string abbreviation of "word"; when not empty it is used in the menu instead of "word"
+--- @field menu? string extra text for the popup menu, displayed after "word" or "abbr"
+--- @field info? string more information about the item, can be displayed in a preview window
+--- @field kind? string single letter indicating the type of completion
+--- @field icase? integer when non-zero case is to be ignored when comparing items to be equal; when omitted zero is used, thus items that only differ in case are added
+--- @field equal? integer when non-zero, always treat this item to be equal when comparing. Which means, "equal=1" disables filtering of this item.
+--- @field dup? integer when non-zero this match will be added even when an item with the same word is already present.
+--- @field empty? integer when non-zero this match will be added even when it is an empty string
+--- @field user_data? any custom data which is associated with the item and available in |v:completed_item|; it can be any type; defaults to an empty string
+--- @field abbr_hlgroup? string an additional highlight group whose attributes are combined with |hl-PmenuSel| and |hl-Pmenu| or |hl-PmenuMatchSel| and |hl-PmenuMatch| highlight attributes in the popup menu to apply cterm and gui properties (with higher priority) like strikethrough to the completion items abbreviation
```

Comment:
This was more about correct typing of the fields, than the description to me, which are a bit lengthy. Could add a short form or emit it entirely.

Reply:
Better to define this as a table and defer the `@field` insertion to the rendering scripts.

Alternatively you can define `vim.v.completed_item` in a source file in `runtime/vim/lua/_meta/*` and just refer to it here. That's what we've done for some of the API keysets.